src/main/java/com/fasterxml/jackson/databind/deser/DeserializerCache.java

```diff
@@ -511,7 +511,7 @@ private JavaType modifyTypeByAnnotation(DeserializationContext ctxt,
                 if (cdDef != null) {
                     JsonDeserializer<?> cd = null;
                     if (cdDef instanceof JsonDeserializer<?>) {
-                        cdDef = (JsonDeserializer<?>) cdDef;
+                        cd = (JsonDeserializer<?>) cdDef;
```

Comment:
I think this may be a legitimate bug

Reply:
Yes, definitely looks like one.

Reply:
That would also qualify for path against 2.11 fwtw.

Reply:
... and the reason it's not found is I suppose that kinds of bugs possible may have been worked around by some other handling.

Might just make sense to leave only against 2.12, since there's no reproduction yet.